### PR TITLE
Add Combination Sum in C++

### PR DESCRIPTION
**Pull request by mgncodes**
Resolves issue #1364 .

### CODE REVIEW

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/* C++ Code for Combination Sum Problem.
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:1]:
```suggestion
/* Combination Sum Problem.
```

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Problem Statement:
+Given an array of integers and a target value, the unique combination of values in array should sum up to the target value. 
+The same number can be chosen multiple times.
+
+For Example:
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:6]: No need to add constraints and example here

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+Constraints:
+1. The array should have distinct elements.
+2. 1 <= target <= 500
+3. 1 <= arr.size() <= 50
+4. 1 <= arr[i] <= 200
+*/
+#include<bits/stdc++.h>
+using namespace std;
+
+void combinationSum(vector<int> &arr, int target, vector<vector<int>> &ans, vector<int> &combn, int beg)
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:19]: add appropriate comments everywhere

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            cout << "\n[";
+            for(int j=0; j<ans[i].size(); ++j) 
+                if(j == ans[i].size() - 1) cout << ans[i][j]; 
+                else cout << ans[i][j] << ", ";
+            cout << "]";
+        }
+    }
+    return 0;
+}
+/* Sample Input
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:58]:
```suggestion
/* Test Cases
```

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Enter Array Size: 4
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:59]: Follow 

Input
Output

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+Enter Array Elements: 1 2 3 4
+Enter Target Sum: 10
+
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:62]: Add 2 TestCases

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by mgncodes — COMMENTED** [review]
Changes done @raksha009

File: Algorithm/Special_algo/Combination_Sum.cpp (hunk 1 above)
Comment (mgncodes) [reply Algorithm/Special_algo/Combination_Sum.cpp]: Done

File: Algorithm/Special_algo/Combination_Sum.cpp (hunk 2 above)
Comment (mgncodes) [reply Algorithm/Special_algo/Combination_Sum.cpp]: Ok Removed

File: Algorithm/Special_algo/Combination_Sum.cpp (hunk 3 above)
Comment (mgncodes) [reply Algorithm/Special_algo/Combination_Sum.cpp:10]: done

File: Algorithm/Special_algo/Combination_Sum.cpp (hunk 4 above)
Comment (mgncodes) [reply Algorithm/Special_algo/Combination_Sum.cpp]: done

File: Algorithm/Special_algo/Combination_Sum.cpp (hunk 5 above)
Comment (mgncodes) [reply Algorithm/Special_algo/Combination_Sum.cpp:70]: done

File: Algorithm/Special_algo/Combination_Sum.cpp (hunk 6 above)
Comment (mgncodes) [reply Algorithm/Special_algo/Combination_Sum.cpp]: added extra testcase

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            cout << "\n[";
+            for(int j=0; j<ans[i].size(); ++j) 
+                if(j == ans[i].size() - 1) cout << ans[i][j]; 
+                else cout << ans[i][j] << ", ";
+            cout << "]";
+        }
+    }
+    return 0;
+}
+/* Sample Input 1:
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:69]:
```suggestion
/* 
Test Cases
Input :
Output : 
```

Floow this

File: Algorithm/Special_algo/Combination_Sum.cpp [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+Sample Input 2:
+Enter Array Size: 4
+Enter Array Elements: 4 2 8 6
+Enter Target Sum: 8
+Output: 
+[2, 2, 2, 2] 
+[2, 2, 4]
+[2, 6]
+[4, 4]
+[8]
```
Comment (raksha009) [inline Algorithm/Special_algo/Combination_Sum.cpp:106]: Add time and space complexities here. Also, do specify the meaning of variable used alongside

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by plazzy99** [comment]
Closing pr due to inactivity

**Pull request closed without merging**